redbot/cogs/admin/converters.py

```diff
@@ -7,15 +7,33 @@
 
 class SelfRole(commands.Converter):
     async def convert(self, ctx: commands.Context, arg: str) -> discord.Role:
-        admin = ctx.command.cog
+        admin = ctx.command.cog.bot.get_cog("Admin")
         if admin is None:
             raise commands.BadArgument(_("The Admin cog is not loaded."))
 
+        selfroles = await admin.config.guild(ctx.guild).selfroles()
         role_converter = commands.RoleConverter()
-        role = await role_converter.convert(ctx, arg)
 
-        selfroles = await admin.config.guild(ctx.guild).selfroles()
+        pool = []
+        for role_id in selfroles:
+            role = ctx.guild.get_role(role_id)
```

Comment:
Since there can be quite a few role I'd personally suggest using AsyncIter here, yes won't make much of a diff but it's good to get in the habit of using it for medium/large loops that does a lot of processing